src/main/java/org/jabref/gui/preferences/ExternalTab.java

```diff
@@ -244,6 +257,10 @@ public void setValues() {
     public void storeSettings() {
         prefs.put(JabRefPreferences.EMAIL_SUBJECT, emailSubject.getText());
         prefs.putBoolean(JabRefPreferences.OPEN_FOLDERS_OF_ATTACHED_FILES, openFoldersOfAttachedFiles.isSelected());
+        if (pushToApplicationComboBox.getValue().getApplicationName() != JabRefPreferences.PUSH_TO_APPLICATION) {
```

Comment:
Similarily, I would extract part of this code to `setActivePushToApplication(PushToApplication)` method in `JabRefPreferences`.